packages/rest/test/unit/coercion/paramStringToInteger.unit.ts

```diff
@@ -18,7 +19,102 @@ const INT64_PARAM = {
   schema: {type: 'integer', format: 'int64'},
 };
 
+const REQUIRED_INTEGER_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {type: 'integer'},
+  required: true,
+};
+
+const INTEGER_PARAM = {
+  in: <ParameterLocation>'path',
+  name: 'aparameter',
+  schema: {type: 'integer'},
+};
+
 describe('coerce param from string to integer', () => {
   test(INT32_PARAM, '100', 100);
   test(INT64_PARAM, '9223372036854775807', 9223372036854775807);
 });
+
+describe('coerce param from string to integer - required', function() {
+  context('valid values', () => {
+    test(REQUIRED_INTEGER_PARAM, '0', 0);
+    test(REQUIRED_INTEGER_PARAM, '1', 1);
+    test(REQUIRED_INTEGER_PARAM, '-1', -1);
+  });
+
+  context('empty values trigger ERROR_BAD_REQUEST', () => {
+    // null, '' sent from request are converted to raw value ''
+    test(
+      REQUIRED_INTEGER_PARAM,
+      '',
+      RestHttpErrors.missingRequired(REQUIRED_INTEGER_PARAM.name),
+    );
+  });
+});
+
+describe('coerce param from string to integer - optional', function() {
+  context('valid values', async () => {
+    test(INTEGER_PARAM, '0', 0);
+    test(INTEGER_PARAM, '1', 1);
+    test(INTEGER_PARAM, '-1', -1);
+  });
+
+  context(
+    'integers larger than MAX_SAFE_INTEGER should trigger ERROR_BAD_REQUEST',
+    async () => {
+      const err = RestHttpErrors.exceedsMaxSafeInt(REQUIRED_INTEGER_PARAM.name);
+      test(INTEGER_PARAM, '2343546576878989879789', err);
+      test(INTEGER_PARAM, '-2343546576878989879789', err);
+      test(INTEGER_PARAM, '1.234e+30', err);
+      test(INTEGER_PARAM, '-1.234e+30', err);
+    },
+  );
+
+  context('scientific notations', () => {
+    test(INTEGER_PARAM, '1.234e+3', 1.234e3);
+    test(INTEGER_PARAM, '-1.234e+3', -1.234e3);
+  });
+
+  context('integer-like string values should trigger ERROR_BAD_REQUEST', () => {
+    test(
+      INTEGER_PARAM,
+      '1.2',
+      RestHttpErrors.invalidData('1.2', INTEGER_PARAM.name),
+    );
+    test(
+      INTEGER_PARAM,
+      '-1.2',
+      RestHttpErrors.invalidData('-1.2', INTEGER_PARAM.name),
+    );
+  });
+
+  context('empty collection converts to undefined', () => {
+    // [], {} sent from request are converted to raw value undefined
+    test(INTEGER_PARAM, undefined, undefined);
+  });
+
+  context('empty values trigger ERROR_BAD_REQUEST', () => {
+    // null, '' sent from request are converted to raw value ''
+    test(INTEGER_PARAM, '', RestHttpErrors.invalidData('', INTEGER_PARAM.name));
+  });
+
+  context(
+    'all other non-integer values should trigger ERROR_BAD_REQUEST',
+    () => {
+      // 'false', false, 'true', true, 'text' sent from request are converted to a string
```

Comment:
Is that the behavior in lb3?
From your comment, I'm guessing 'false' will always be coerced into false?

Reply:
@shimks yep see https://github.com/strongloop/strong-remoting/blob/master/test/rest-coercion/jsonform-integer.suite.js

In strong-remoting, they are e2e tests: coercion for response data before http client processes it, and in lb4 the unit tests are coercion for response data after it's passed through http client, so those values are converted to a string.